### PR TITLE
pin v1.16 ahash to 0.8.3

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

poor upstream release management

#### Summary of Changes

pin ahash

### CODE REVIEW

**Comment by t-nelson** [comment]
ohhh i think it's because these test cargo.lock aren't under version control. the error comes from resolution of the index, to build the lockfile. mebe we can just commit them...

File: sdk/cargo-build-sbf/tests/crates/fail/Cargo.lock [hunk 1]
```diff
@@ -0,0 +1,1516 @@
+# This file is automatically @generated by Cargo.
+# It is not intended for manual editing.
```
Comment (t-nelson) [inline sdk/cargo-build-sbf/tests/crates/fail/Cargo.lock:2]: and yet, here i am!

**Comment by t-nelson** [comment]
@dmakarov on a scale of 1-10, how bad do you hate https://github.com/solana-labs/solana/pull/34650/commits/3ad21cd8d702bac7439a6cf876476e67ce5e30ee ?

**Comment by dmakarov** [comment]
> @dmakarov on a scale of 1-10, how bad do you hate [3ad21cd](https://github.com/solana-labs/solana/commit/3ad21cd8d702bac7439a6cf876476e67ce5e30ee) ?

0. I don't mind this change at all. It shouldn't affect the tests it seems.

**Review by willhickey — APPROVED** [review]

**Comment by t-nelson** [comment]
>     0. I don't mind this change at all. It shouldn't affect the tests it seems.

great, thanks!

**Comment by willhickey** [comment]
If I understand correctly this solves the problem without impacting the actual build. If that's correct I'd prefer this over 34645 to reduce risk for [the v1.16 releases we need to bridge to v1.17](https://discord.com/channels/428295358100013066/910937142182682656/1192329666216874074)

**Review by yihau — APPROVED** [review]
🔥

**Comment by t-nelson** [comment]
> If I understand correctly this solves the problem without impacting the actual build. If that's correct I'd prefer this over 34645 to reduce risk for [the v1.16 releases we need to bridge to v1.17](https://discord.com/channels/428295358100013066/910937142182682656/1192329666216874074)

correct

**Comment by yihau** [comment]
bummer 🫠 I got this one.
Screenshot 2024-01-09 at 21 42 12

**Comment by t-nelson** [comment]
this is from secondary?

**Comment by CriesofCarrots** [comment]
ahash v0.8.3 was yanked, so pinning to that specific version isn't going to work. v0.8.4 is probably the most-reasonable alternative, given how many old versions have been yanked.

**Comment by yihau** [comment]
> this is from secondary?

yes

**Comment by yihau** [comment]
I'm thinking that maybe we don't need to pin the ahash version for all crates. we only pin the ahash version for programs. (keep sdk/cargo-build-sbf/tests/crates/fail/Cargo.lock and sdk/cargo-build-sbf/tests/crates/noop/Cargo.lock).

the original error originated from `test_build`. If users import our crates and not for a Solana program, they probably shouldn't be restricted by this 🤔 

btw, I just checked the build log of v1.16.23. It seems that we have already compiled ahash 0.8.6 to crates. so I guess if we still prefer to pin ahash, 0.8.4 won't be so dangerous. (even 0.8.6)

**Comment by yihau** [comment]
I'm going to ping you in two different PRs for this issue 🪖

1. #34723
2. #34716

**Pull request merged**